lineman/app/components/group_page/join_group_button/join_group_button.coffee

```diff
@@ -0,0 +1,23 @@
+angular.module('loomioApp').directive 'joinGroupButton', ->
+  scope: {group: '='}
+  restrict: 'E'
+  templateUrl: 'generated/components/group_page/join_group_button/join_group_button.html'
+  replace: true
+  controller: ($scope, AbilityService, ModalService, CurrentUser, Records, FlashService, MembershipRequestForm) ->
+    Records.membershipRequests.fetchMyPendingByGroup($scope.group.key)
+
+    $scope.isMember = ->
+      CurrentUser.membershipFor($scope.group)?
```

Comment:
I like how that reads too.. However canJoinGroup is something the join_group_button will need to know.. 

````
%join_group_button
%request_membership_button
````

Can feel a wee bit loopy but is also eaiser to use. The pattern of nf-if on the inside becomes more necessary when we look at the memebership_requests_card. It needs to fire a request to see if there are requests before you know if it should display or not. If you have ng-if on the outside, then you need to know about requesting membership_requests on the group_page_controller... yuck